broker/graphite/src/macro_cache.cc

```diff
@@ -36,7 +35,7 @@ macro_cache::macro_cache(const std::shared_ptr<persistent_cache>& cache)
     std::shared_ptr<io::data> d;
     do {
       _cache->get(d);
-      write(d);
+      write(neb::bbdo2_to_bbdo3(d));
```

Comment:
Same here as for bam.
I would prefer the function called only when needed, so if the event is an old one, we change it to a new one.
So calling this bbdo2_to_bbdo3() in the switch case when needed.